From e9ea9a0f8aa390f9357dee2bc854474fc51534e0 Mon Sep 17 00:00:00 2001
From: Cheick TRAORE <cheick.traore-ext@st.com>
Date: Tue, 4 Jun 2024 08:40:11 +0200
Subject: [PATCH 0943/1695] drivers: counter: stm32_lptimer: add support for
 stm32mp21

Add support for stm32mp21 in stm32_lptimer driver by using a new
compatible.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7bda838ae8df17a59e51890c7a4b5d32afb8fd42 drivers: counter: stm32_lptimer: add support for stm32mp21]
Change-Id: Ida926a57942942a2d205ddfb3ffa820ae1037780
Signed-off-by: Cheick TRAORE <cheick.traore-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/384917
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/counter/stm32_lptimer.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/core/drivers/counter/stm32_lptimer.c b/core/drivers/counter/stm32_lptimer.c
index f612b3556..fe077d009 100644
--- a/core/drivers/counter/stm32_lptimer.c
+++ b/core/drivers/counter/stm32_lptimer.c
@@ -491,6 +491,7 @@ static TEE_Result stm32_lptimer_set_driverdata(struct lptimer_device *lpt_dev)
 #ifdef CFG_EMBED_DTB
 static const char * const stm32_lptimer_counter_compat[] = {
 	"st,stm32-lptimer-counter",
+	"st,stm32mp21-lptimer-counter",
 	"st,stm32mp25-lptimer-counter",
 };
 
@@ -674,6 +675,10 @@ static const struct dt_device_match stm32_lptimer_match_table[] = {
 		.compatible = "st,stm32-lptimer",
 		.compat_data = &lptimer_cdata,
 	},
+	{
+		.compatible = "st,stm32mp21-lptimer",
+		.compat_data = &lptimer_mp25_cdata,
+	},
 	{
 		.compatible = "st,stm32mp25-lptimer",
 		.compat_data = &lptimer_mp25_cdata,
-- 
2.47.2

